storage/src/mmr/journaled.rs

```diff
@@ -250,12 +250,30 @@ impl<E: RStorage + Clock + Metrics, H: CHasher> Mmr<E, H> {
 
     /// Add an element to the MMR and return its position in the MMR. Elements added to the MMR
     /// aren't persisted to disk until `sync` is called.
+    ///
+    /// # Warning
+    ///
+    /// Panics if there are unprocessed updates.
     pub async fn add(&mut self, h: &mut impl Hasher<H>, element: &[u8]) -> Result<u64, Error> {
         self.mem_mmr.add(h, element).await
     }
 
+    /// Add an element to the MMR, delaying the computation of ancestor digests
+    /// until the next `sync`.
+    pub async fn add_batched(
```

Comment:
nit: do you think there should be a dedicated test here for this new function (just to make sure calling self.mem_mmr.sync behaves as expected)?

I suppose this is well-tested by the things living on top but 🤷

Reply:
🤦 yes of course, added.